### PR TITLE
Speed up of installed pip pkgs

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
For 9 pkgs: 2.5 -> 0.5 sec

### CODE REVIEW

**Comment by ar13pit** [comment]
Haha nice. I'll look at this on Friday.

File: installer/tue-install-impl.bash [hunk 1]
```diff
@@ -866,20 +866,45 @@ function _tue-install-pip-now
...
+        else
+            pips_to_check="$pips_to_check $pkg"
+        fi
+    done
+
+    read -r -a pips_to_check <<< "$pips_to_check"
+    local installed_versions
+    installed_versions=$(python"${pv}" "$TUE_INSTALL_SCRIPTS_DIR"/check-pip-pkg-installed-version.py "${pips_to_check[@]}")
+    local error_code=$?
+    if [ "$error_code" -gt 1 ]
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply installer/tue-install-impl.bash:887]: Yep, probably. But execution shouldn't continue

File: installer/tue-install-impl.bash [hunk 2]
```diff
@@ -866,20 +866,45 @@ function _tue-install-pip-now
...
+    then
+        tue-install-error "tue-install-pip${pv}-now: $installed_versions"
+    fi
+    read -r -a installed_versions <<< "$installed_versions"
+
+    if [ "${#pips_to_check[@]}" -ne "${#installed_versions[@]}" ]
+    then
+        tue-install-error "Lengths of pips_to_check, ${#pips_to_check[@]}, and installed_version, ${#installed_versions[@]}, don't match"
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply installer/tue-install-impl.bash:895]: True

File: installer/tue-install-impl.bash [hunk 3]
```diff
@@ -866,20 +866,45 @@ function _tue-install-pip-now
...
             git_pips_to_install="$git_pips_to_install $pkg"
-        elif ! installed_version=$(python"${pv}" "$TUE_INSTALL_SCRIPTS_DIR"/check-pip-pkg-installed-version.py "$pkg")
+        else
+            pips_to_check="$pips_to_check $pkg"
+        fi
+    done
+
+    read -r -a pips_to_check <<< "$pips_to_check"
+    local installed_versions
+    installed_versions=$(python"${pv}" "$TUE_INSTALL_SCRIPTS_DIR"/check-pip-pkg-installed-version.py "${pips_to_check[@]}")
```
Comment (Timple) [inline installer/tue-install-impl.bash:885]: There might not be any pips to check (all git+ notation). Then this command fails because of no input.

**Pull request merged**